source/core/index.ts

```diff
@@ -674,6 +678,7 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 			response.destroy();
 
 			this._beforeError(new ReadError(error, this));
+			this.removeListener('abort', this.abortHandler);
```

Comment:
`response` is not guaranteed, but `request` is. However listening on `error` request event is also incorrect because of retries. I'd suggest to move this to `_destroy`.